lib/cucumber/core/compiler.rb

```diff
@@ -38,8 +38,11 @@ def create_test_case(pickle)
         uri = pickle.uri
         test_steps = pickle.steps.map { |step| create_test_step(step, uri) }
         lines = source_lines_for_pickle(pickle).sort.reverse
+        location = Test::Location.new(uri, lines)
```

Comment:
If these are becoming procedural lets try isolate a few into something a bit more private. I'm only briefly here for a few days though so I might be wrong

Reply:
@luke-hill I think you're right. I just pushed another commit moving more of these details into private methods. If we move to a kwargs constructor for `Test::Case` and `Test::Step` we can reduce this even further by omitting the variable assignment, which is mostly for constructor arg readability.